tests/end-to-end/Smoke.Tests/Utilities/MSBuildFixture.cs

```diff
@@ -16,7 +17,9 @@ public void RegisterInstance()
         {
             if (Interlocked.Exchange(ref _registered, 1) == 0)
             {
-                MSBuildHelpers.HookAssemblyResolveForMSBuild();
+                // During testing we just need a default MSBuild instance to be registered.
+                var defaultInstance = MSBuildLocator.QueryVisualStudioInstances().First();
+                MSBuildHelpers.HookAssemblyResolveForMSBuild(defaultInstance.MSBuildPath);
```

Comment:
Specify a MSBuildPath otherwise the smoketests will pause while prompting to select a MSBuild instance to use.